File: Java/_06_extras/Max.java
package _06_extras;

// Berechne die Maximaltemperatur mithilfe einer Lambda Funktion und ohne eine von Java bereitgestellte Funktion

import java.util.List;

public class Max {
    public static void main(String[] args) {
        List<Integer> temperatures_celsius = List.of(25, 30, 22, 18, 28, 35, 20);

        int max = temperatures_celsius.stream()
                    .reduce(0, (x, y) -> {
                        if (x > y) return x;
                        return y;
                    });

        System.out.println(max);
    }
}

File: Python/06_extras/max.py
# Berechne die Maximaltemperatur mithilfe einer Lambda Funktion und ohne eine von Python bereitgestellte Funktion

from functools import reduce

temperatures_celsius = [25, 30, 22, 18, 28, 35, 20]

highest_temperature = reduce(lambda x, y: x if x > y else y, temperatures_celsius)
print(highest_temperature)